### PR TITLE
Now handles stripping of .so/.dll files in windows/linux.

### PR DESCRIPTION
**Pull request by dcdillon**
Since I can't test Solaris, I don't try to strip and let the debugging symbols bloat the shared library.

Anyhow, the crux of it is we have no `src/Makevars` file.  Just two files `src/Makevars.gcc` and `src/Makevars.solaris`.  Analogously we have two files that define the `.onLoad` function (`inst/onLoad/onLoad.gcc.R` and `inst/onLoad/onLoad.solaris.R`).  The appropriate one is copied in to the `src` or `R` directory appropriately.

`src/Makevars.gcc` defines `PKG_LIBS=-Wl,-S`.  `src/Makevars.solaris` does not.

Both of the `onLoad` files define a `.onLoad` function that registers an `RcppHoney` plugin with Rcpp.  The Solaris version does nothing.  The gcc version defines `libs=-Wl,-S`.

Then when a user wants to use `RcppHoney` in a source file they need to add the following two lines to the `.cpp` file:

``` cpp
// [[Rcpp::depends(RcppHoney)]]
// [[Rcpp::plugin(RcppHoney)]]
```

After this, things, "just work".

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Interesting.  You think they'll let that on?   Plugin is a very clever hack to get that in there.

**Comment by eddelbuettel** [comment]
(Oh, and do you want me to merge, or will you once you slept over it?)

**Comment by dcdillon** [comment]
Go ahead and merge.

**Comment by dcdillon** [comment]
if you think it's good...it passes `--as-cran`

**Comment by eddelbuettel** [comment]
What about clang and OS X and all that?  What about Windoze?

**Comment by dcdillon** [comment]
Works fine for Windoze.  clang and gcc share command line syntax so it's fine there.  I think it's good...might break on AlpineLinux, but I'll cross that bridge when I come to it.

**Comment by dcdillon** [comment]
I guess the only thing I question is whether I should move `src/Makevars.gcc` and `src/Makevars.solaris` to `inst`.

Thoughts?

**Comment by eddelbuettel** [comment]
Before copying them back in?  Could also be  `src/platform/` and then copy up ...

`inst/` isn't needed -- it ensure everything below it survives into the installed package.  This should only be needed in the source package.

**Comment by dcdillon** [comment]
I just stuck stuff in `inst` because I knew it wouldn't flag it as "bad" there. OK...gimme a sec...will move to `/src/platform`.

**Comment by dcdillon** [comment]
Any thoughts on where I should put the `onLoad.gcc.R` and `onLoad.solaris.R` files?

**Comment by dcdillon** [comment]
Aside from the obvious "I'll tell you where you can put them..."

**Comment by dcdillon** [comment]
DON'T MERGE.  CHANGE IMMINENT.

**Comment by dcdillon** [comment]
New code per [Rcpp#522](https://github.com/RcppCore/Rcpp/issues/522).  Now if we just get the folks at [Rcpp](https://github.com/RcppCore/Rcpp) to adopt the `strip` attribute like I've done the `RcppHoney` attribute, we won't have to have ANY special `configure` file and things will "just work".

File: configure [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MYOS=`uname -s`
+
+echo $MYOS
+
+if [ "$(echo "$MYOS" | cut -c1-5)" = "SunOS" ]; then
+	cp src/platform/Makevars.solaris src/Makevars
+else
+	cp src/platform/Makevars.gcc src/Makevars
+fi
```
Comment (eddelbuettel) [inline configure:9]: Very nice.

**Comment by dcdillon** [comment]
Going to merge this.  When we get the Rcpp attribute strip, we can get rid of configure.

**Pull request merged**